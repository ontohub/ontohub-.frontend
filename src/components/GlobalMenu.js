import React from 'react'
import { Link } from 'react-router-dom'
import { Menu } from 'semantic-ui-react'
import { css, sizes, colors } from '../styles'

const styles = css({
  position: 'fixed',
  backgroundColor: colors.dark,
  width: '100%',
  top: 0,
  display: 'flex',
  justifyContent: 'center'
})

const menuStyles = css({
  height: sizes.menuHeight,
  width: `${sizes.contentWidth}px !important`,
  position: 'relative !important'
})

const GlobalMenu = () => (
  <div {...styles}>
    <Menu inverted borderless fixed="top" {...menuStyles}>
      <Menu.Item header as={Link} to="/">Ontohub</Menu.Item>
    </Menu>
  </div>
)

export default GlobalMenu
