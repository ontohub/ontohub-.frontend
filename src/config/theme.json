{
  "colors": {
    "dark": "#1B1C1D",
    "light": "rgba(255, 255, 255, 0.9)"
  },
  "sizes": {
    "contentWidth": "700px",
    "headerPadding": "30px",
    "menuHeight": "50px",
    "footerMargin": "100px"
  }
}
