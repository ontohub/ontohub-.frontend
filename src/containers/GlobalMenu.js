import { currentUserQuery, signInMutation } from '../apollo/queries'
import { compose, graphql, withApollo } from 'react-apollo'
import { GlobalMenu } from '../components'
import { userValidations } from '../helpers/validations'
import _ from 'lodash'

const signOut = (client) => () => {
  localStorage.removeItem('auth-token')
  return client.resetStore()
}
const signIn = (client, token) => {
  localStorage.setItem('auth-token', token)
  client.resetStore()
}

const GlobalMenuWithData = compose(
  withApollo,
  graphql(currentUserQuery, {
    props: (props) => {
      if (
        !_.get(props, 'data.error.networkError') &&
        (props.data.error || (!props.data.me && !props.data.loading))
      ) {
        // Invalid user token
        signOut(props.ownProps.client)()
      }
      return {
        ...props.ownProps,
        loading: _.get(props, 'data.loading'),
        error: _.get(props, 'data.error'),
        me: _.get(props, 'data.me')
      }
    }
  }),
  graphql(signInMutation, {
    props: (props) => ({
      ...props.ownProps,
      client: undefined,
      signUpValidations: userValidations(props.ownProps.client),
      onSignIn: (username, password) =>
        props
          .mutate({
            variables: { username, password }
          })
          .then((response) => {
            if (response.data.signIn) {
              return signIn(props.ownProps.client, response.data.signIn.token)
            } else {
              throw new Error('Sign in failed')
            }
          }),
      onSignOut: signOut(props.ownProps.client)
    })
  })
)(GlobalMenu)

export default GlobalMenuWithData
