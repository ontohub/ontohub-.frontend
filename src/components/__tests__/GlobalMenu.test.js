import React from 'react'
import GlobalMenu from '../GlobalMenu'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { ThemeProvider } from 'styled-components'
import theme from '../../styles'

describe('GlobalMenu', () => {
  describe('signed out user', () => {
    it('matches the snapshot', () => {
      const wrapper = render(
        <Router>
          <ThemeProvider
            theme={{ colors: { dark: '#000' }, sizes: { menuHeight: '50px' } }}
          >
            <GlobalMenu />
          </ThemeProvider>
        </Router>
      )
      expect(toJSON(wrapper)).toMatchSnapshot()
    })
  })
  describe('signed in user', () => {
    it('matches the snapshot', () => {
      const wrapper = render(
        <Router>
          <ThemeProvider
            theme={{ colors: { dark: '#000' }, sizes: { menuHeight: '50px' } }}
          >
            <GlobalMenu
              me={{ id: 'ada', emailHash: '3e3417d7ef77d5932a6734b916515ed5' }}
            />
          </ThemeProvider>
        </Router>
      )
      expect(toJSON(wrapper)).toMatchSnapshot()
    })
  })
  it('contains the title link', () => {
    const wrapper = render(
      <Router>
        <ThemeProvider theme={theme}>
          <GlobalMenu />
        </ThemeProvider>
      </Router>
    )
    expect(wrapper.find('a').text()).toContain('Ontohub')
  })
})
