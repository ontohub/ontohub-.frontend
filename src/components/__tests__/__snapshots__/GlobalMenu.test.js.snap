// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalMenu matches the snapshot 1`] = `
<div
  data-css-poy59o=""
>
  <div
    className="ui borderless inverted top fixed menu"
    data-css-k2eqqe=""
  >
    <a
      className="header item"
      href="/"
      onClick={[Function]}
    >
      Ontohub
    </a>
  </div>
</div>
`;
